crates/ibc/src/clients/ics07_tendermint/client_state.rs

```diff
@@ -860,9 +855,13 @@ impl TryFrom<RawTmClientState> for ClientState {
         // In `RawClientState`, a `frozen_height` of `0` means "not frozen".
         // See:
         // https://github.com/cosmos/ibc-go/blob/8422d0c4c35ef970539466c5bdec1cd27369bab3/modules/light-clients/07-tendermint/types/client_state.go#L74
-        let frozen_height = raw
+        if raw
             .frozen_height
-            .and_then(|raw_height| raw_height.try_into().ok());
+            .and_then(|h| Height::try_from(h).ok())
+            .is_some()
+        {
+            return Err(Error::FrozenHeightNotAllowed);
+        }
```

Comment:
I don't think this is the right approach; the domain type should be allowed to be created as frozen. What we want in this PR is to disallow a client to be already frozen in the context of the `CreateClient` handler (see other comment as to whether we actually want that).

If we were to want this behavior, I think it should go in `ClientState::initialise()`

Reply:
given the latest [comment](https://github.com/cosmos/ibc-rs/pull/602#issuecomment-1504050779), let's move forward with this change since ibc-go will also adopt the change. Thus, only this applies:

> If we were to want this behavior, I think it should go in `ClientState::initialise()`

Reply:
Not sure if I feel you correctly:

- Initializing an instance of a client happens through [`decode_client_state`](https://github.com/cosmos/ibc-rs/blob/b10f4baea722f3942f81fdeb610c84ad4e664f92/crates/ibc/src/core/ics02_client/handler/create_client.rs#L68) call. And with the current implementation, we won’t end up creating a Tm client with frozen height through that process. Of course, before that, we neither pass a raw ClientState with non-frozen height.
- Plus, from what I see, `intialise` method serves to return a `TmConsensusState`, and considering the [recent spec change](https://github.com/cosmos/ibc/commit/ba9c4f82e0c706761e5b4be5a4fbc270357e09e1#diff-41a81331fa27dcef562c77b90b8068c2bf506add8735c89a822eb04659734322R172), `initialize` would not deal with returning a `ClientState` anymore and mainly handle setting the states.
- And, the needed checks that exist inside `initialize` there in spec, are basically covered through `ClientState::new` call on our side